core/src/main/java/com/cloud/storage/template/HttpTemplateDownloader.java

```diff
@@ -218,7 +219,10 @@ public long download(boolean resume, DownloadCompleteCallback callback) {
             errorString = hte.getMessage();
```

Comment:
Why not set it if errorString is null as below?

Reply:
> Why not set it if errorString is null as below?

It may be an easy code change, but then the scope of this issue will be greater, which will require more testing as well.